### PR TITLE
Add Swaps-based prices table "tokenprices"

### PR DESCRIPTION
**Pull request by tonykau1**
Script to backfill liquidity pools and tokens tables.  This uses harmony API to grab the token0, token1, name, symbol, and decimals based on the token_address.

tokenprices table generated from swaps pricing transactions and the liquidity_pools table.

liquidity_pools table requires tokens table to be updated with both token0 and token1.  Updating tokens table is currently a manual process (via backfill script run on a list of tokens)

### CODE REVIEW

File: models/core/liquidity_pools.sql
Comment (antonyip) [inline models/core/liquidity_pools.sql]:
```suggestion
where event_name = 'PairCreated'
```

File: models/core/liquidity_pools.sql
Comment (antonyip) [inline models/core/liquidity_pools.sql]: any reason for the extra concat white space in the front?

File: models/core/liquidity_pools.sql
Comment (antonyip) [inline models/core/liquidity_pools.sql]:
```suggestion
  inner join {{ ref('tokens') }} as t0
```

File: models/core/liquidity_pools.sql
Comment (antonyip) [inline models/core/liquidity_pools.sql]:
```suggestion
  inner join {{ ref('tokens') }} as t1
```

File: models/staging/stg_tokenprice_from_swaps.sql
Comment (antonyip) [inline models/staging/stg_tokenprice_from_swaps.sql]: I think we can add the incremental_last_x_days macro here. Be sure to test it tho...

File: models/staging/stg_tokenprice_from_swaps.sql
Comment (antonyip) [inline models/staging/stg_tokenprice_from_swaps.sql]:
```suggestion
select * from final

```

File: models/staging/stg_tokenprice_from_swaps.yml
Comment (antonyip) [inline models/staging/stg_tokenprice_from_swaps.yml:63]:
```suggestion
          
          
```

File: models/staging/stg_tokenprice_from_swaps.yml
Comment (antonyip) [inline models/staging/stg_tokenprice_from_swaps.yml:55]: add unique test

File: models/tokenprice/tokenprices.sql
Comment (antonyip) [inline models/tokenprice/tokenprices.sql]:
```suggestion
  select
```

File: models/tokenprice/tokenprices.sql
Comment (antonyip) [inline models/tokenprice/tokenprices.sql:13]: think we can have the incremental macro here

File: models/tokenprice/tokenprices.sql
Comment (antonyip) [inline models/tokenprice/tokenprices.sql]:
```suggestion
select * from final

```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
    description: "Volume-weighted average daily prices based on swap data, assuming curated stables = 1 USD, and preferring the most liquid pair.  Token prices will show up if they have at least USD100 of daily volume."
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
        description: The date of the token price.
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
        description: Address of the Token.
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml:43]:
```suggestion
          
          
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
        description: HRC token symbol.
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
        description: Volume-weighted average price calculated on the `block_date`.
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
        description: The USD-equivalent amount swapped on the `block_date` based on curated liquidity pool list.
```

File: models/tokenprice/tokenprices.yml
Comment (antonyip) [inline models/tokenprice/tokenprices.yml]:
```suggestion
        description: The amount of tokens swapped on this date based on the curated liquidity pool list.
```

**Review by antonyip — COMMENTED** [review]
Some minor changes needed. Overall great job! Love the script to extract tokens and LP. Will approve after fixes. 

@chuxinh can do another round of checking if you want to.

File: models/tokenprice/tokenprices.sql
Comment (antonyip) [inline models/tokenprice/tokenprices.sql:4]:
```suggestion
        unique_key='block_date || token_address',,
```

Comment (antonyip) [reply models/tokenprice/tokenprices.sql:4]: needed this to get it to work.

Comment (antonyip) [inline models/tokenprice/tokenprices.sql]:
```suggestion
        unique_key="key",
```
somehow this fails on dbt cloud for incremental builds

File: models/tokenprice/tokenprices.sql
Comment (antonyip) [inline models/tokenprice/tokenprices.sql:20]:
```suggestion
    key,
    block_date,
```
in conjunction with the unique_key being added to the incremental

File: models/core/liquidity_pools.sql
Comment (tonykau1) [reply models/core/liquidity_pools.sql]: Without the space in front of the 'LP', there was no space between token1 and "LP".  Because multiple concats, I changed the line to the below to make it more readable:

t0.token_symbol || '-' || t1.token_symbol || ' LP' as pool_name, -- "TOKEN0-TOKEN1 LP"

**Review by antonyip — APPROVED** [review]
LGTM! Will give it a few days on my dev branch to see if anything breaks before merging.

**Pull request merged**
